src/Components/src/Table/AnnotationTableContextMenu.fs

```diff
@@ -15,19 +28,132 @@ type ATCMC =
         {| element = Html.kbd [ prop.className "swt:ml-auto swt:kbd swt:kbd-sm"; prop.text text ]; label = label|}
 
 type AnnotationTableContextMenuUtil =
+
     static member clear (tableIndex: CellCoordinate, cellIndex: (int * int), table: ArcTable, selectHandle: SelectHandle) : ArcTable =
         if selectHandle.contains tableIndex then
             table.ClearSelectedCells(selectHandle)
         else
             table.ClearCell(cellIndex)
         table.Copy()
 
+    static member deleteRow (tableIndex: CellCoordinate, cellIndex: (int * int), table: ArcTable, selectHandle: SelectHandle) : ArcTable =
+        if selectHandle.contains tableIndex then
+            let lastCellCoordinate =
+                selectHandle.getSelectedCells()
+                |> Array.ofSeq
+                |> Array.last
+
+            table.RemoveRow(lastCellCoordinate.y - 1)
+        else
+            table.RemoveRow(fst cellIndex)
+        table.Copy()
+
+    static member deleteColumn (tableIndex: CellCoordinate, cellIndex: (int * int), table: ArcTable, selectHandle: SelectHandle) : ArcTable =
+        if selectHandle.contains tableIndex then
+            let lastCellCoordinate =
+                selectHandle.getSelectedCells()
+                |> Array.ofSeq
+                |> Array.last
+
+            table.RemoveColumn(lastCellCoordinate.x - 1)
+        else
+            table.RemoveRow(fst cellIndex)
+        table.Copy()
+
+    static member copy (cellIndex: (int * int), table: ArcTable, selectHandle: SelectHandle) =
+
+        let result =
+            if selectHandle.getCount() > 1 then
+
+                let cellCoordinates =
+                    selectHandle.getSelectedCells()
+                    |> Array.ofSeq
+                    |> Array.groupBy (fun item -> item.y)
+
+                let cells =
+                    cellCoordinates
+                    |> Array.map (fun (_, row) ->
+                        row
+                        |> Array.map (fun coordinate ->
+                            table.GetCellAt(coordinate.x - 1, coordinate.y - 1)
+                        )
+                    )
+
+                CompositeCell.ToTableTxt(cells)
+            else
+                let cell = table.GetCellAt(cellIndex)
+                cell.ToTabStr()
+
+        navigator.clipboard.writeText result
+        |> Promise.start
+
+    static member paste ((columnIndex, rowIndex): (int * int), table: ArcTable, selectHandle: SelectHandle, setTable) =
```

Comment:
This is incredibly complex. I am not yet sure how to correctly handle this. As we already got requests about only pasting values for unitized cells for example (keep unit, only change value).

It is very dofficult to read this part on phone. Can you give me a summary of the paste logic?